src/HieDb/Create.hs

```diff
@@ -400,35 +430,27 @@ removeDependencySrcFiles (getConn -> conn) =
 
 {-| Delete all occurrences of given @.hie@ file from the database -}
 deleteFileFromIndex :: HieDb -> FilePath -> IO ()
-deleteFileFromIndex (getConn -> conn) path = withTransaction conn $ do
-  deleteInternalTables conn path
+deleteFileFromIndex db@(getConn -> conn) path = withTransaction conn $ do
+  deleteInternalTablesStatement db (Only path)
 
 {-| Delete all entries associated with modules for which the 'modInfoSrcFile' doesn't exist
 on the disk.
 Doesn't delete it if there is no associated 'modInfoSrcFile'
 -}
 deleteMissingRealFiles :: HieDb -> IO ()
-deleteMissingRealFiles (getConn -> conn) = withTransaction conn $ do
+deleteMissingRealFiles db@(getConn -> conn) = withTransaction conn $ do
   missing_file_keys <- fold_ conn "SELECT hieFile,hs_src FROM mods WHERE hs_src IS NOT NULL AND is_real" [] $
     \acc (path,src) -> do
       exists <- doesFileExist src
       pure $ if exists then acc else path : acc
   forM_ missing_file_keys $ \path -> do
-    deleteInternalTables conn path
+    deleteInternalTablesStatement db (Only path)
 
 {-| Garbage collect typenames with no references - it is a good idea to call
 this function after a sequence of database updates (inserts or deletes)
 -}
 garbageCollectTypeNames :: HieDb -> IO Int
 garbageCollectTypeNames (getConn -> conn) = do
   execute_ conn "DELETE FROM typenames WHERE NOT EXISTS ( SELECT 1 FROM typerefs WHERE typerefs.id = typenames.id LIMIT 1 )"
+  execute_ conn "PRAGMA optimize;"
   changes conn
-
-deleteInternalTables :: Connection -> FilePath -> IO ()
```

Comment:
Same as above - probably not a good idea to change public api (unless we want to do major version bump, which I don't think is necessary). 

https://hackage-content.haskell.org/package/hiedb-0.7.0.0/docs/HieDb-Create.html#v:deleteInternalTables

Reply:
Approximately the same problem occurs here as with `initConn`. In this case I could keep this function for API purposes and have it keep calling the non-prepared deletions, but otherwise not call it ourselves in the library? And also perhaps decorate it with a DEPRECATED pragma?